File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) return false;
        Map<Character, Integer> m= new HashMap<>();
        for(int i=0;i<s.length();i++){
            m.put(s.charAt(i),m.getOrDefault(s.charAt(i),0)+1);
        }

        for(int i=0;i<t.length();i++){
            if(!m.containsKey(t.charAt(i)) || m.get(t.charAt(i)) == 0) return false;
            m.put(t.charAt(i),m.getOrDefault(t.charAt(i),0)-1);
        }
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) return false;
        // map<char,int> freq;
        // for (auto i: s){
        //     freq[i]++;
        // }
        // for(auto it : t){
        //     if(freq.find(it) == freq.end() || freq[it] == 0) return false;
        //     freq[it]--;
        // }
        // return true;

        sort(s.begin(), s.end());
        sort(t.begin(),t.end());
        if(s != t) return false;
        return true;
    }
};